alchemiscale/tests/integration/storage/test_statestore.py

```diff
@@ -277,10 +440,69 @@ def test_queue_task(self, n4js: Neo4jStore, network_tyk2, scope_test):
         network_sk2 = n4js.create_network(an2, scope_test)
         taskqueue_sk2: ScopedKey = n4js.create_taskqueue(network_sk2)
 
-        with pytest.raises(ValueError, match="not found in same network"):
-            task_sks_fail = n4js.queue_taskqueue_tasks(task_sks, taskqueue_sk2)
+        task_sks_fail = n4js.action_tasks(task_sks, taskqueue_sk2)
+        assert all([i is None for i in task_sks_fail])
+
+    def test_cancel_task(self, n4js, network_tyk2, scope_test):
+        an = network_tyk2
+        network_sk = n4js.create_network(an, scope_test)
+        taskqueue_sk: ScopedKey = n4js.create_taskqueue(network_sk)
+
+        transformation = list(an.edges)[0]
+        transformation_sk = n4js.get_scoped_key(transformation, scope_test)
+
+        # create 10 tasks
+        task_sks = [n4js.create_task(transformation_sk) for i in range(10)]
+
+        # queue the tasks
+        actioned = n4js.action_tasks(task_sks, taskqueue_sk)
+
+        # cancel the second and third task in the queue
+        canceled = n4js.cancel_tasks(task_sks[1:3], taskqueue_sk)
+
+        # check that the queue has the contents we expect
+        tasks = n4js.graph.run(
+            f"""
+                MATCH (tq:TaskQueue {{_scoped_key: '{taskqueue_sk}'}})-->
+                      (head:TaskQueueHead)<-[:FOLLOWS* {{taskqueue: '{taskqueue_sk}'}}]-(task:Task)
+                return task
+                """
+        )
+        tasks = [record["task"] for record in tasks]
+
+        assert len(tasks) == 8
+        assert set([ScopedKey.from_str(t["_scoped_key"]) for t in tasks]) == set(
+            actioned
+        ) - set(canceled)
+
+        # check the order is preserved
```

Comment:
I think ordering in this case will become non-deterministic when #78 is in? No change needs to be made but I may need to modify the test.

Reply:
Indeed, we'll ditch the ordering tests here for #78/#80.